### PR TITLE
SNAT parsing

### PR DESCRIPTION
**Pull request by Adphi**

### CODE REVIEW

File: parser_test.go [hunk 1]
```diff
@@ -706,6 +706,26 @@ func TestParser_Parse(t *testing.T) {
 			},
 			err: nil,
 		},
+		{
+			name: "parse rule with jump target SNAT",
+			s:    "-A foo -o eth0 -4 -j SNAT --to-source 192.168.1.1",
```
Comment (leonnicolas) [inline parser_test.go:711]: can you add a test case where `--to-source` is not the last flag?

File: parser_test.go [hunk 2]
```diff
@@ -959,13 +979,15 @@ func TestParser_Parse(t *testing.T) {
...
 		},
 	} {
-		p := NewParser(strings.NewReader(tc.s))
-		s, err := p.Parse()
-		if !reflect.DeepEqual(tc.r, s) {
-			t.Errorf("%d. %s: %q result mismatch:\n\texp=%v\n\tgot=%v\n\terr=%v", i, tc.name, tc.s, tc.r, s, err)
-		} else if tc.err != err && tc.err.Error() != err.Error() {
-			t.Errorf("%d. %s: %q error mismatch:\n\texp=%v\n\tgot=%v.", i, tc.name, tc.s, tc.err, err)
-		}
+		t.Run(tc.name, func(t *testing.T) {
```
Comment (leonnicolas) [inline parser_test.go:1003]: thanks, that is very helpful

**Comment by leonnicolas** [comment]
The SNAT target supports more flags than `--to-source`. I think we should add them before merging. Here is the part of the man pages:
```
 SNAT
       This  target  is  only valid in the nat table, in the POSTROUTING and INPUT chains, and user-defined chains which are only called from those chains.  It specifies that the source address of the packet
       should be modified (and all future packets in this connection will also be mangled), and rules should cease being examined.  It takes the following options:

       --to-source [ipaddr[-ipaddr]][:port[-port]]
              which can specify a single new source IP address, an inclusive range of IP addresses. Optionally a port range, if the rule also specifies one of the following protocols: tcp, udp, dccp or sctp.
              If no port range is specified, then source ports below 512 will be mapped to other ports below 512: those between 512 and 1023 inclusive will be mapped to ports below 1024, and other ports will
              be mapped to 1024 or above. Where possible, no port alteration will occur.  In Kernels up to 2.6.10, you can add several --to-source options. For those kernels, if you  specify  more  than  one
              source  address,  either  via  an  address  range  or  multiple  --to-source  options,  a simple round-robin (one after another in cycle) takes place between these addresses.  Later Kernels (>=
              2.6.11-rc1) don't have the ability to NAT to multiple ranges anymore.

       --random
              If option --random is used then port mapping will be randomized through a hash-based algorithm (kernel >= 2.6.21).

       --random-fully
              If option --random-fully is used then port mapping will be fully randomized through a PRNG (kernel >= 3.14).

       --persistent
              Gives a client the same source-/destination-address for each connection.  This supersedes the SAME target. Support for persistent mappings is available from 2.6.29-rc2.
```

**Comment by Adphi** [comment]
> The SNAT target supports more flags than --to-source. I think we should add them before merging. 

@leonnicolas I only implemented what I needed, but I completely agree with you. I'll take a look as soon as I can.

File: parser_test.go (hunk 1 above)
Comment (Adphi) [reply parser_test.go:711]: It's done 😉

**Comment by leonnicolas** [comment]
Thanks @Adphi

**Review by leonnicolas — APPROVED** [review]

**Comment by Adphi** [comment]
Thank you @leonnicolas !

**Pull request merged**